woocommerce-services.php

```diff
@@ -530,11 +530,29 @@ public function init_core_wizard_shipping_config() {
 
 		public function init_core_wizard_payments_config() {
 			$stripe_settings = get_option( 'woocommerce_stripe_settings', false );
-			$stripe_enabled  = is_array( $stripe_settings )
+
+			$should_create_stripe_account  = is_array( $stripe_settings )
 				&& ( isset( $stripe_settings['create_account'] ) && 'yes' === $stripe_settings['create_account'] )
 				&& ( isset( $stripe_settings['enabled'] ) && 'yes' === $stripe_settings['enabled'] );
 
-			if ( $stripe_enabled && is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' ) ) {
+			// In certain scenarios, the user can enter an email address but not connect Jetpack in the wizard,
+			// but instead add the Stripe keys manually and connect Jetpack after. If the existing keys are detected,
+			// forget the wizard settings and never retry
+			$stripe_already_connected = is_array( $stripe_settings )
+				&& (
+					! empty( $stripe_settings['test_publishable_key'] )
+					|| ! empty( $stripe_settings['test_secret_key'] )
+					|| ! empty( $stripe_settings['publishable_key'] )
+					|| ! empty( $stripe_settings['secret_key'] )
+				);
+			if ( $should_create_stripe_account && $stripe_already_connected ) {
+				unset( $stripe_settings['email'] );
```

Comment:
Why is `$stripe_settings['email']` getting deleted? I don't see it deleted when going through with the Stripe account creation.

Reply:
> I think once we determine that we don't need the wizard options anymore, we should clean up everything.

Agreed.

So I looked it up and the Stripe extension does not using the `email` setting at all, for at least the last 3 versions. Same with WooCommerce itself.

The reason the `email` option got introduced is for this wizard set up. I don't think that Stripe should rely on that, but if you want to double-check with the team, though, that seems reasonable, too.

Given this, I don't think it will cause problems if we delete it. It would be great to be able to test it while developing though, so if you want to delete it in a separate PR I'm game for that as well, so it doesn't go out in the next release.

Reply:
Cool - shall we also delete `email` in the case that we create the account for the user as well then? Within this `if`:

https://github.com/Automattic/woocommerce-services/blob/dfb4904468ee663cf3431117649cd0623f84b814/woocommerce-services.php#L555